rest_sensors.cpp

```diff
@@ -1364,6 +1364,24 @@ int DeRestPluginPrivate::changeSensorConfig(const ApiRequest &req, ApiResponse &
                     {
                         mfrCode = VENDOR_JENNIC;
                         attrId = 0x4003;
+
+                        // Setting the heat setpoint disables off/boost modes,
+			// but this is not reported back by the thermostat.
+			// Hence, the off/boost flags will be removed here to reflect the actual operating state.
+                        if (hostFlags == 0)
+                        {
+                            const NodeValue &val = sensor->getZclValue(THERMOSTAT_CLUSTER_ID, 0x4008);
+                            hostFlags = val.value.u32;
+                        }
+
+			hostFlags &= 0xffffeb;
```

Comment:
https://github.com/dresden-elektronik/deconz-rest-plugin/issues/1098#issuecomment-462077343

Please add a small comment  to annote the related flags, after a few months it's hard to tell what they do :)

// clear set off (0x10) and boost (0x04) flags
hostFlags &= ~(0x10 | 0x04);

Reply:
you're right, it is more clear this way. Done :)